src/Bravo3/ImageManager/Services/ImageManager.php

```diff
@@ -188,18 +188,20 @@ public function pull(Image $image)
     /**
      * Get meta information about the source image from the cache layer.
      *
-     * @param Image $image
+     * @param Image|string $image
      *
      * @return ImageMetadata
      */
-    public function getMetadata(Image $image)
+    public function getMetadata($image)
     {
         // If the $image is a variation, refer to its parent
         // for the metadata.
         if ($image instanceof ImageVariation) {
             $img_key = $image->getKey(true);
```

Comment:
Ah, I see your point now. Still, if you pass a key, this should retrieve the metadata for that exact key. Not try and go looking for that keys parent key. Does that make sense?